src/NHibernate/Async/Engine/CascadingAction.cs

```diff
@@ -0,0 +1,334 @@
+			{
+				if (cancellationToken.IsCancellationRequested)
+				{
+					return Task.FromCanceled<object>(cancellationToken);
+				}
+				try
+				{
+					if (log.IsDebugEnabled)
+					{
+						log.Debug("cascading to refresh: " + entityName);
+					}
+					return session.RefreshAsync(child, (IDictionary)anything, cancellationToken);
+				}
+				catch (Exception ex)
+				{
+					return Task.FromException<object>(ex);
+				}
+			}
+		}
+
+		/// <content>
+		/// Contains generated async methods
+		/// </content>
+		private partial class EvictCascadingAction : CascadingAction
+		{
+			public override Task CascadeAsync(IEventSource session, object child, string entityName, object anything, bool isCascadeDeleteEnabled, CancellationToken cancellationToken)
+			{
+				if (cancellationToken.IsCancellationRequested)
+				{
+					return Task.FromCanceled<object>(cancellationToken);
+				}
+				try
+				{
+					if (log.IsDebugEnabled)
+					{
+						log.Debug("cascading to evict: " + entityName);
+					}
+					return session.EvictAsync(child, cancellationToken);
+				}
+				catch (Exception ex)
+				{
+					return Task.FromException<object>(ex);
+				}
+			}
+		}
+
+		/// <content>
+		/// Contains generated async methods
+		/// </content>
+		private partial class SaveUpdateCascadingAction : CascadingAction
+		{
+			public override Task CascadeAsync(IEventSource session, object child, string entityName, object anything, bool isCascadeDeleteEnabled, CancellationToken cancellationToken)
+			{
+				if (cancellationToken.IsCancellationRequested)
+				{
+					return Task.FromCanceled<object>(cancellationToken);
+				}
+				try
+				{
+					if (log.IsDebugEnabled)
+					{
+						log.Debug("cascading to saveOrUpdate: " + entityName);
+					}
+					return session.SaveOrUpdateAsync(entityName, child, cancellationToken);
+				}
+				catch (Exception ex)
+				{
+					return Task.FromException<object>(ex);
+				}
+			}
+		}
+
+		/// <content>
+		/// Contains generated async methods
+		/// </content>
+		private partial class MergeCascadingAction : CascadingAction
+		{
+			public override Task CascadeAsync(IEventSource session, object child, string entityName, object anything, bool isCascadeDeleteEnabled, CancellationToken cancellationToken)
+			{
+				if (cancellationToken.IsCancellationRequested)
+				{
+					return Task.FromCanceled<object>(cancellationToken);
+				}
+				try
+				{
+					if (log.IsDebugEnabled)
+					{
+						log.Debug("cascading to merge: " + entityName);
+					}
+					return session.MergeAsync(entityName, child, (IDictionary)anything, cancellationToken);
+				}
+				catch (Exception ex)
+				{
+					return Task.FromException<object>(ex);
+				}
+			}
+		}
+        
+		/// <content>
+        /// Contains generated async methods
+        /// </content>
+        private partial class PersistCascadingAction : CascadingAction
```

Comment:
Formatting glitch maybe due to synchronous code having some bad white-spaces (8 spaces) on the seemingly empty line preceding this class.